pyrdp/security/nla.py

```diff
@@ -32,16 +34,36 @@ def __init__(self, sink: IntermediateLayer, state: NTLMSSPState, log: logging.Lo
         super().__init__()
         self.sink = sink
         self.ntlmSSPState = state
-        self.log = log
         self.ntlmSSPParser = NTLMSSPParser()
+        self.ntlmCapture = ntlmCapture
+        self.log = log
+
+    def getRandChallenge(self):
+        """
+        Generate a random 32-bit challenge
+        """
+        challenge = b'%016x' % randbits(16 * 4)
+        return decode(challenge, 'hex')
 
     def onUnknownHeader(self, header, data: bytes):
-        messageOffset = self.ntlmSSPParser.findMessage(data)
+        signatureOffset = self.ntlmSSPParser.findMessage(data)
 
-        if messageOffset != -1:
-            message: NTLMSSPPDU = self.ntlmSSPParser.parse(data[messageOffset :])
+        if signatureOffset != -1:
+            message: NTLMSSPPDU = self.ntlmSSPParser.parse(data)
             self.ntlmSSPState.setMessage(message)
 
+            if message.messageType == NTLMSSPMessageType.NEGOTIATE_MESSAGE and self.ntlmCapture:
+                randomChallenge = self.getRandChallenge()
+                self.log.info("NTLMSSP Negotiation")
```

Comment:
Since it doesn't add value or content from the client, I think this should be changed to a `debug` level message instead of `info`. `info` level message would generate a json event for that log but I don't think it would be valuable.